scripts/mbedtls_dev/build_tree.py

```diff
@@ -21,6 +21,14 @@ def looks_like_mbedtls_root(path: str) -> bool:
 def looks_like_root(path: str) -> bool:
     return looks_like_tf_psa_crypto_root(path) or looks_like_mbedtls_root(path)
 
+def crypto_core_directory() -> str:
+    if looks_like_tf_psa_crypto_root(os.path.curdir):
```

Comment:
Not the current directory: that only works if the current directory is the repo root. The repo root needs to be passed as an argument. It would make sense to make the argument optional, and if it's absent, use the value of `guess_mbedtls_root()`.

Python language note: optional means
```
def crypto_core_directory(root: Optional[str] = None) -> str:
    if root is None:
        root = guess_repo_root()
```
Not `crypto_core_directory(root: str = guess_repo_root())` because that would call `guess_repo_root()` at the time the function is defined rather than at the time the function is called.

Reply:
agree with @gilles-peskine-arm proposal

Reply:
Note for @gilles-peskine-arm and @ronald-cron-arm. I have made the above change but the function `guess_repo_root` doesn't exist. So I initially made the change calling the function `guess_mbedtls_root` and I tacked another commit on the end where I replace the name `guess_mbedtls_root` with `guess_project_root`. I chose this name for consistency. Additionally the final commit replaces all calls to `guess_mbedtls_root` with the new name. I did it this way so that if people don't like this change then I can simply axe the last commit.